src/tools/run-make-support/src/lib.rs

```diff
@@ -271,6 +271,34 @@ pub fn set_host_rpath(cmd: &mut Command) {
     });
 }
 
+/// Read the contents of a file that cannot simply be read by
+/// read_to_string, due to invalid utf8 data, then assert that it contains `expected`.
+#[track_caller]
+pub fn invalid_utf8_contains<P: AsRef<Path>>(path: P, expected: &str) {
+    let buffer = fs_wrapper::read(path.as_ref());
+    if !String::from_utf8_lossy(&buffer).contains(expected) {
+        eprintln!("=== FILE CONTENTS ===");
```

Comment:
Suggestion: this should remark that the display is lossy w.r.t. UTF-8. Something like "FILE CONTENTS (LOSSY)"